internal/controller/service.go

```diff
@@ -113,13 +124,26 @@ func (s *Service) HandlePermission(ctx context.Context, ar *api.AccessRequest, a
 
 // handleAccessExpired will remove the Argo CD access for the subject and
 // update the AccessRequest status field.
-func (s *Service) handleAccessExpired(ctx context.Context, ar *api.AccessRequest, rt *api.RoleTemplate) error {
+func (s *Service) handleAccessExpired(ctx context.Context, ar *api.AccessRequest, app *argocd.Application, rt *api.RoleTemplate) error {
+	log := log.FromContext(ctx)
+	statusDetails := ""
+	if s.hasPlugin() {
+		resp, err := s.accessRequester.RevokeAccess(ar, app)
+		if err != nil {
+			log.Error(err, "Error invoking plugin RevokeAccess function")
+			return fmt.Errorf("error invoking plugin RevokeAccess function: %w", err)
```

Comment:
Logging an error immediately before returning it often lead to duplicate error message in logs. Only return the error and let the caller decide how to handle the error.

Reply:
fixed